src/apps/HeyGPT/HeyGPT/ViewModels/LetsChatViewModel.cs

```diff
@@ -120,6 +137,74 @@ private void OnExecuteToggleShouldIncludeClipboardContextCommand()
 
     #endregion
 
+    #region Event Handlers
+
+    private void SubscribeToEvents()
+    {
+        _eventAggregator.Subscribe<FileActivatedEvent>(OnFileActivatedEvent);
+        _eventAggregator.Subscribe<ProtocolActivatedEvent>(OnProtocolActivatedEvent);
+    }
+
+    private void OnFileActivatedEvent(FileActivatedEvent fileActivatedEvent)
+    {
+        var firstFile = fileActivatedEvent.Files[0];
+
+        var messagePrompt = GetMessagePromptFromFile(firstFile);
+
+        if (messagePrompt is null)
+        {
+            return;
+        }
+
+        var message = messagePrompt.Value.Message;
+        var character = GetChatCharacterDetailsFromString(messagePrompt.Value.Character);
+
+        if (!string.IsNullOrWhiteSpace(message))
+        {
+            App.MainWindow.DispatcherQueue.TryEnqueue(DispatcherQueuePriority.Normal, async () =>
```

Comment:
I'm not thrilled with the direct reference to App.MainWindow... We need to either abstract the dispatcher with a service or use a library that will abstract away the need to Enqueue the DispatcherQueue manually.   In the past we've used may approaches, and its never just straightforward :)